finance-admin-api/api/request_report.go

```diff
@@ -32,13 +33,12 @@ func (s *Server) requestReport(w http.ResponseWriter, r *http.Request) error {
 		}
 	}
 
-	go func() {
-		ctx := context.Background()
-		err := s.generateAndUploadReport(ctx, reportRequest, time.Now())
+	go func(logger *slog.Logger) {
+		err := s.generateAndUploadReport(context.Background(), reportRequest, time.Now())
 		if err != nil {
-			telemetry.LoggerFromContext(ctx).Error(err.Error())
+			logger.Error(err.Error())
 		}
-	}()
+	}(telemetry.LoggerFromContext(r.Context()))
```

Comment:
Is there a reason for this change? I haven't used goroutines enough to know the benefit of this approach.

Reply:
Basically it reduces the risk of an error - before, where we were creating the logger inside the goroutine using ctx from outside the goroutine, there was a risk of the context being closed before the logger was created. By passing in the logger, created from context, when we create the goroutine, we can assure that context is not closed when the logger is created, and that the goroutine will have access to that logger.